### DISCUSSION TITLE
Optional fields in FormModel not working

### DISCUSSION THREAD

**Discussion by D1maD1m0nd** [fastapi/fastapi#13077] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class FashionSwapFormData(BaseModel):
    PersonImage: Optional[UploadFile] = File(
        default=None
    ),
    PersonUrl: Optional[str] = Form(
        default=None
    ),

    ClothesImage: Optional[UploadFile] = File(
        default=None
    ),
    ClothesUrl: Optional[str]= Form(
        default=None
    ),
#not working
@router_generation.post(
    "/fashion_swap/",
    response_model_exclude_none=True,
)
async def fashion_swap(
        data: Annotated[FashionSwapFormData, Form(media_type="multipart/form-data")],
) 

#working
@router_generation.post(
    "/fashion_swap/",
    response_model_exclude_none=True,
)
async def fashion_swap(
     PersonImage: Optional[UploadFile] = File(
        default=None
    ),
    PersonUrl: Optional[str] = Form(
        default=None
    ),

    ClothesImage: Optional[UploadFile] = File(
        default=None
    ),
    ClothesUrl: Optional[str]= Form(
        default=None
    ),
)
```

### Description

Hello, in the code above I do not take into account opional fields in the form model, but if I specify them in the function in the same way, everything works fine

For unfilled fields I get the following error

> {
    "detail": [
        {
            "type": "string_type",
            "loc": [
                "body",
                "PersonUrl"
            ],
            "msg": "Input should be a valid string",
            "input": [
                {
                    "embed": null,
                    "media_type": "application/x-www-form-urlencoded",
                    "example": null,
                    "include_in_schema": true,
                    "openapi_examples": null
                }
            ]
        },
        {
            "type": "value_error",
            "loc": [
                "body",
                "ClothesImage"
            ],
            "msg": "Value error, Expected UploadFile, received: <class 'tuple'>",
            "input": [
                {
                    "embed": null,
                    "media_type": "multipart/form-data",
                    "example": null,
                    "include_in_schema": true,
                    "openapi_examples": null
                }
            ],
            "ctx": {
                "error": {}
            }
        }
    ]
}

### Operating System

Windows

### Operating System Details

windows 11 

### FastAPI Version

0.115.6

### Pydantic Version

2.10.3

### Python Version

3.12.7

**Comment by yvallois** [comment, marked as answer]
when I copied and pasted your code my ide transformed your model into:
```python
class FashionSwapFormData(BaseModel):
    PersonImage: Optional[UploadFile] = (File(default=None),)
    PersonUrl: Optional[str] = (Form(default=None),)

    ClothesImage: Optional[UploadFile] = (File(default=None),)
    ClothesUrl: Optional[str] = (Form(default=None),)
```
I had the same error as you


But if you remove the comma
```python
class FashionSwapFormData(BaseModel):
    PersonImage: Optional[UploadFile] = File(default=None)
    PersonUrl: Optional[str] = Form(default=None)
    ClothesImage: Optional[UploadFile] = File(default=None)
    ClothesUrl: Optional[str] = Form(default=None)
```
every thing works fine
